Marlin/src/module/planner.cpp

```diff
@@ -2769,49 +2769,66 @@ bool Planner::_populate_block(
      * Heavily modified. Originally adapted from Průša firmware.
      * https://github.com/prusa3d/Prusa-Firmware
      */
-    #ifndef TRAVEL_EXTRA_XYJERK
-      #define TRAVEL_EXTRA_XYJERK 0.0f
+    #ifdef TRAVEL_EXTRA_XYJERK
+      #define HAS_TRAVEL_EXTRA_XYJERK 1
+      const float extra_xyjerk = TERN0(HAS_EXTRUDERS, dist.e <= 0) ? TRAVEL_EXTRA_XYJERK : 0.0f;
     #endif
-    const float extra_xyjerk = TERN0(HAS_EXTRUDERS, dist.e <= 0) ? TRAVEL_EXTRA_XYJERK : 0.0f;
 
     if (!moves_queued || UNEAR_ZERO(previous_nominal_speed)) {
       // Compute "safe" speed, limited by a jerk to/from full halt.
 
       float v_factor = 1.0f;
       LOOP_LOGICAL_AXES(i) {
-        const float jerk = ABS(current_speed[i]),   // Starting from zero, change in speed for this axis
-                    maxj = max_jerk[i] + (i == X_AXIS || i == Y_AXIS ? extra_xyjerk : 0.0f); // The max jerk setting for this axis
+        const float jerk = ABS(current_speed[i]);   // Starting from zero, change in speed for this axis
+        float maxj = max_jerk[i];
+        TERN_(HAS_TRAVEL_EXTRA_XYJERK, if (i == X_AXIS || i == Y_AXIS) maxj += extra_xyjerk);
         if (jerk * v_factor > maxj) v_factor = maxj / jerk;
```

Comment:
We can technically save at least one and up to `LOGICAL_AXES` mults here and in line 2832 by doing 
```c++
if (v_factor < 1.0f) {
  if (jerk > maxj) v_factor = maxj / jerk;
}
else {
  if (jerk * v_factor> maxj) v_factor = maxj / jerk;
}
```

On systems with an FPU, this is probably close to +-0 cycles, but on an AVR, this should save a couple. On the downside, it adds a bit of clutter to the code.

Reply:
It's also the extra maintenance/potential pitfalls. Example your pre-edit version had a < instead of ==. Benchmarking, I'm getting ~665 before vs ~766 cycles after. I was expecting maybe a small increase due to the extra comparison every cycle but not that much. I'm guessing compiler is probably doing a (v_factor - 1.0f) == 0 but still. Not sure if it's worth putting more time into this unless there's someone that's willing to benchmark on AVR.

Reply:
If it's actually slower on Arm/Cortex, I also don't see any point in pursuing this anymore. Thanks for checking.